depends/patches/qt/fix_android_jni_static.patch

```diff
@@ -0,0 +1,18 @@
+--- old/qtbase/src/plugins/platforms/android/androidjnimain.cpp
++++ new/qtbase/src/plugins/platforms/android/androidjnimain.cpp
+@@ -890,6 +890,14 @@
```

Comment:
Can you add some more info here about what this patch is doing / link to somewhere upstream if applicable.

Reply:
When JVM loads a native lib (via [system.load](https://developer.android.com/reference/java/lang/System#load(java.lang.String))) it invokes its `JNI_OnLoad` function. Qt for Android uses this mechanism to do all kinds of init and register native functions for the Java side to use. The proper and supported way is to dynamically link and then each separate `libqt*.so` inits itself through `JNI_OnLoad` as it is loaded.

However I wanted to link statically into one big blob so with this tiny patch I basically combined the only two Qt Base module `JNI_OnLoad` calls into one. At runtime goes swimmingly and the GUI loads.

This dependency on `JNI_OnLoad` for init and the linker confusion it causes is perhaps something that can be improved upstream. I haven't seen it documented anywhere except in [QTBUG-32618](https://bugreports.qt.io/browse/QTBUG-32618).